python/cudf/dataframe/index.py

```diff
@@ -702,7 +712,12 @@ def as_index(arbitrary, name=None):
         - GenericIndex for all other inputs.
     """
     # This function should probably be moved to Index.__new__
+    if hasattr(arbitrary, 'name') and name is None:
+        name = arbitrary.name
+
     if isinstance(arbitrary, Index):
+        if name is not None:
+            arbitrary.name = name
```

Comment:
You can't do this because you're now modifying the input object instead of creating a new object.

Reply:
@kkraus14 So, shall I create a new Index object with updated name in this case?

Reply:
In that case we can combine this case with last else case.

Reply:
Yes

Reply:
or create a copy of arbitrary object and modify it's name and return it?

Reply:
Looks like we have a `rename` exclusively for that - Used it.